go/metadata/merger_test.sh

```diff
@@ -16,11 +16,43 @@
 #
 ################################################################################
 #
+set +e
 printenv
 
+error=0
+
 $TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/merger --output $TEST_TMPDIR/out.json \
 	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/chrome-linux.json \
 	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/android-browser-gingerbread-nexus-s.json
 
 diff $TEST_TMPDIR/out.json \
-	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/merger-result.json
\ No newline at end of file
+	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/merger-result.json
+if [ $? != 0 ]; then
+	echo "Merge result didn't match."
+	error=1
+fi
+
+$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/merger --output $TEST_TMPDIR/out2.json \
+	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/named-files1.json \
+	$TEST_SRCDIR/$TEST_WORKSPACE/go/metadata/testdata/named-files1.json
+if [ $? != 0 ]; then
+	echo "Expected successful named files merge failed."
```

Comment:
I find this error message hard to parse. This is the intent, right?
> [I] expected [a] successful named-files-merge [but I tried it and it] failed

How about more explicit statements of what you expect?
* Merging named files maps with conflicting names should fail.
* bad-named-files.json should fail to load.